src/registration_and_login/LoginForm.js

```diff
@@ -0,0 +1,86 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import validator from 'validator';
+import { isEmail, isLength } from 'validator';
+
+const LoginForm = () => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [isValidEmail, setIsValidEmail] = useState(true);
+  const [error, setError] = useState('');
+  const [status, setStatus] = useState('Not loged');
+
+  const handleEmailChange = (event) => {
+    const newEmail = event.target.value;
+    setEmail(newEmail);
+    setIsValidEmail(validator.isEmail(newEmail));
+  };
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+
+      // Perform validation
+  if (!isEmail(email)) {
+    setError('Please enter a valid email address.');
+    return;
+  }
+
+  if (!isLength(password, { min: 6 })) {
+    setError('Password should be at least 6 characters long.');
+    return;
+  }
+
+  // Continue with login process
+  // You can make an API call here to authenticate the user
+
+  // Clear form fields and error
+  setEmail('');
+  setPassword('');
+  setError('');
+
+    try {
+      const response = await axios.get(`${process.env.REACT_APP_API_ROOT_VAR}/api/users/login?email=${email}&password=${password}`);
+      setStatus(response.data);
```

Comment:
Tu, a potom dole, keď robíš `status === 'Login successful'` sa príliš spoliehaš na správu z API a tým príliš preväzuješ tento kód s kódom API (tight coupling). API by mala ideálne indikovať úspech alebo neúspech loginu cez response status code 200 pre úspech a 401 pre neplatné heslo a užívateľské meno, ale ja som si to musel tiež vygoogliť a vedú sa o tom spory:

- https://stackoverflow.com/questions/11714485/restful-login-failure-return-401-or-custom-response
- https://stackoverflow.com/questions/32752578/whats-the-appropriate-http-status-code-to-return-if-a-user-tries-logging-in-wit

API sú väčšinou chránené pomocou OAuth 2.0 protokolu, ale to ti musím vysvetliť osobne.